airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/SyncWorkflowImpl.java

```diff
@@ -4,32 +4,39 @@
 
 package io.airbyte.workers.temporal.sync;
 
-import io.airbyte.config.Configs;
-import io.airbyte.config.EnvConfigs;
 import io.airbyte.config.NormalizationInput;
 import io.airbyte.config.NormalizationSummary;
 import io.airbyte.config.OperatorDbtInput;
-import io.airbyte.config.ResourceRequirements;
 import io.airbyte.config.StandardSyncInput;
 import io.airbyte.config.StandardSyncOperation;
 import io.airbyte.config.StandardSyncOperation.OperatorType;
 import io.airbyte.config.StandardSyncOutput;
 import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
 import io.airbyte.scheduler.models.IntegrationLauncherConfig;
 import io.airbyte.scheduler.models.JobRunConfig;
-import io.airbyte.workers.temporal.scheduling.shared.ActivityConfiguration;
-import io.temporal.activity.ActivityOptions;
+import io.airbyte.workers.temporal.annotations.TemporalActivityStub;
 import io.temporal.workflow.Workflow;
 import java.util.UUID;
+import javax.inject.Singleton;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+@Singleton
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
 public class SyncWorkflowImpl implements SyncWorkflow {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(SyncWorkflowImpl.class);
   private static final String VERSION_LABEL = "sync-workflow";
   private static final int CURRENT_VERSION = 2;
-  private static final int PREV_VERSION = 1;
+
+  @TemporalActivityStub(activityOptionsBeanName = "longRunActivityOptions")
+  private ReplicationActivity replicationActivity;
```

Comment:
I don't know if it is actually an issue but it looks like we are introducing some scoping change here.
If I got the injection right, we are now going to have a single stub per activity for the singleton while the previous code was generating new stubs per run. I remember that temporal is doing some queue inheritance behind the scene, would this change interfere with that process?

For example, I may have missed the right factory, but I am not sure where the `setTaskQueue` logic went.

Reply:
Yes.  The previous code was actually needlessly re-creating the activity per run, which is not something that the [Temporal documentation](https://docs.temporal.io/java/activities/#invoking-activities-inside-workflows) recommends.  They say that you should just create the stub once for the workflow implementation instead of creating it each time the workflow is executed.  Unless we have a need to dynamically change the configuration of the stub per workflow execution (which as far as I can tell, we do not), this is the preferred way to create the stubs.